src/lib/operators/aggregate.cpp

```diff
@@ -727,7 +728,16 @@ void Aggregate::write_aggregate_output(ColumnID column_index) {
   }
 
   // write aggregated values into the column
-  _write_aggregate_values<ColumnType, decltype(aggregate_type), function>(col, context->results);
+  if (!context->results->empty()) {
+    _write_aggregate_values<ColumnType, decltype(aggregate_type), function>(col, context->results);
+  } else if (_groupby_columns.empty()) {
+    // If we did not GROUP BY anything and we have no results, we need to add NULL for most aggregates and 0 for count
+    col->values().push_back(decltype(aggregate_type){});
+    if (function != AggregateFunction::Count && function != AggregateFunction::CountDistinct) {
```

Comment:
Why not also ignore `AggregateFunction::Sum`? I would assume that the sum of an empty column is `0` and not `null`.

Reply:
Ok, I found the answer: https://dba.stackexchange.com/a/25467 Not intuitive, but standard conform.